addons/mail/models/mail_channel.py

```diff
@@ -88,22 +88,30 @@ def _generate_random_token(self):
         ('groups', 'Selected group of users')], string='Privacy',
         required=True, default='groups',
         help='This group is visible by non members. Invisible groups can add members through the invite button.')
-    group_public_id = fields.Many2one('res.groups', string='Authorized Group',
-                                      default=lambda self: self.env.ref('base.group_user'))
+    group_public_id = fields.Many2one('res.groups', string='Authorized Group', compute='_compute_group_public_id', readonly=False, store=True)
 
     _sql_constraints = [
         ('channel_type_not_null', 'CHECK(channel_type IS NOT NULL)', 'The channel type cannot be empty'),
         ('uuid_unique', 'UNIQUE(uuid)', 'The channel UUID must be unique'),
+        ('group_public_id_check',
+         "CHECK (channel_type = 'channel' OR group_public_id IS NULL)",
+         'Group authorization and group auto-subscription are only supported on channels.')
     ]
 
-    # CHAT CONSTRAINT
+    # CONSTRAINTS
 
     @api.constrains('channel_last_seen_partner_ids', 'channel_partner_ids')
     def _constraint_partners_chat(self):
         for ch in self.sudo().filtered(lambda ch: ch.channel_type == 'chat'):
             if len(ch.channel_last_seen_partner_ids) > 2 or len(ch.channel_partner_ids) > 2:
                 raise ValidationError(_("A channel of type 'chat' cannot have more than two users."))
 
+    @api.constrains('group_public_id', 'group_ids')
+    def _constraint_group_id_channel(self):
+        failing_channels = self.sudo().filtered(lambda channel: channel.channel_type != 'channel' and (channel.group_public_id or channel.group_ids))
+        if failing_channels:
+            raise ValidationError(_("channel_type should be 'channel' to have the group-based authorization."))
```

Comment:
add failing channels in the error message to help debugging

Reply:
Also missing "or group auto-subscription" in the message (same idea as the SQL error)